src/addToFavorites.ts

```diff
@@ -0,0 +1,38 @@
+import * as vscode from 'vscode';
+import { FavoritesEntry, FavoritesProvider } from './favorites';
+import { HistoryEntry } from './history';
+
+/**
+ * add a gif to the favorites
+ * @param  {HistoryEntry} gif the gif to add to the favorites
+ * @param  {vscode.Memento} state the state holder in which to get the current favorites to modify them
+ * @param  {FavoritesProvider} favoritesTreeView the favorites tree view that is linked to the favorites menu, that needs to be refreshed for the change to appear to the user
+ */
+export const addGifToFavorites: (
+	gif: HistoryEntry,
+	state: vscode.Memento,
+	favoritesTreeView: FavoritesProvider
+) => void = (
+	gif: HistoryEntry,
+	state: vscode.Memento,
+	favoritesTreeView: FavoritesProvider
+) => {
+	const favorites: FavoritesEntry[] | undefined = state.get('favorites');
+	if (favorites) {
+		if (favorites.filter(entry => entry.gifUri === gif.gifUri).length !== 0) {
+			vscode.window.showInformationMessage(
+				'This GIF was already in your favorites'
+			);
+		} else {
+			const newFav = new FavoritesEntry(gif.label, gif.gifUri);
+			state.update('favorites', [newFav].concat(favorites)).then(() => {
```

Comment:
no limit for favorites? Not saying we should, just wondering if it's a design choice or a missing statement. I'd rather have unlimited favorites if it works

Reply:
yes, I thought that favorites were user controlled, while the history keeps getting bigger